### PR TITLE
Added run private network notes

### PR DESCRIPTION
**Pull request by dgarcia360**
I would like to have a folder with some minimal docs for catapult-server self-contained in the repo.

The idea is to keep these notes updated with new releases so that other devs can try catapult-server directly from the build image and create custom tooling on top to launch their networks.

Please, review the first note under ``docs/RUNNETWORKLIN.md``. If you like the style, I will be submitting more instructions. At the moment I have planned:

* Running a node (connect to an existent network).
* Running an API node (setup and enable MongoDB extension).

Related issue https://github.com/nemtech/symbol-docs/issues/314

### CODE REVIEW

**Comment by dgarcia360** [comment]
> Something not covered, that probably should be, are extension configs for Dual vs API vs Peer. I would reference this repo for building such configurations: https://github.com/nemfoundation/testnet-config

Thanks for your comments @CrackTheCode016! We'll be adding separate instructions to run extra peer, API, and dual nodes. You can see in this other PR https://github.com/nemtech/catapult-server/pull/67 some first instructions to connect a peer node to an existing network. I would appreciate receiving your feedback and some extra tip that might help other developers to build custom tooling around catapult-server.

File: README.md [hunk 1]
```diff
@@ -15,20 +15,20 @@ symbol-server code is organized as follows:
...
+| /sdk | Reusable code used by tests and tools. |
 | /seed | Nemesis blocks used in tests. |
 | /src | Symbol's core engine. |
 | /tests | Collection of tests. |
 | /tools | Tools to deploy and monitor networks and nodes. |
 
 ## Building the Image
 
-To compile symbol-server source code, follow the [building instructions](BUILDING.md). 
+To compile symbol-server source code, follow the [building instructions](docs/BUILDING.md).
```
Comment (Jaguar0625) [inline README.md:31]: catapult-server ? (check for others)

File: README.md [hunk 2]
```diff
@@ -46,9 +46,13 @@ At the time of writing, the main public network has not been launched. Follow th
 
 ### Private Test Network
 
-With Symbol, businesses can launch and extend private networks by developing custom plugins and extensions at the protocol level. The package [Service Bootstrap] contains the necessary setup scripts to deploy a network for testing and development purposes with just one command. 
+With Symbol, businesses can launch and extend private networks by developing custom plugins and extensions at the protocol level.
 
-To run a private test net, follow [this guide](https://nemtech.github.io/guides/network/creating-a-private-test-net.html#creating-a-private-test-net).
+The package [Service Bootstrap] contains the necessary setup scripts to deploy a network for testing and development purposes with just one command.
```
Comment (Jaguar0625) [inline README.md:51]: dead link ?

File: README.md (hunk 1 above)
Comment (Jaguar0625) [inline README.md:31]: catapult-server ? [check for other references]

File: docs/RUNNETWORKLIN.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+The scripts have been tested on servers with the following minimum requirements.
+
+* CPU: 2 cores or more
+* Memory: 4GB or more
+* HD: 20GB or more
+* OS: Ubuntu 18.04 (LTS)
+
+The port ``7900`` is required by catapult-server to communicate between nodes. Make sure that the server's host is accessible from the internet and that the port is open and available.
+
+NOTE: Although you might be able to run the software in less powerful instances, you might encounter some issues while installing or running the node.
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:20]: consider having separate md of HW requirements that is reference by both RUN mds

File: docs/RUNNETWORKLIN.md [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+Prerequisites
+---
+
+* Follow [catapult-server build](BUILDING.md) instructions.
+
+Copy the configuration template
+---
+
+After building catapult-server, copy the configuration templates from the root folder of the repository under the ``_build`` directory.
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:30]: this is generally *NOT* recommended

File: docs/RUNNETWORKLIN.md [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+cd catapult-server/_build
+cp ../resources/* resources/
+cp ../../tools/nemgen/resources/mijin-test.properties ../resources/
+```
+
+Generate accounts
+---
+
+```sh
+./bin/catapult.tools.address -g 10 --network mijin-test > nemesis.addresses.txt
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:42]: should note this is creating a nemesis with 10 accounts, but number of accounts is customizable

File: docs/RUNNETWORKLIN.md [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+2\. Replace ``[cpp]`` and ``[output]`` sections with the following configuration.
+
+```ini
+[cpp]
+
+cppFileHeader = ../../HEADER.inc
+
+[output]
+
+cppFile = ../seed/network-test/MockMemoryBasedStorage_data.h
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:72]: cppFileHeader + cppFile should be set empty to not generate file, which is only needed for certain tests 

```
cppFileHeader = 
...
cppFile =
```

File: docs/RUNNETWORKLIN.md [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+
+NOTE: By default, ``initialCurrencyAtomicUnits`` and ``totalChainImportance`` do not match the values set in ``./resources/mijin-test.properties``.
+Other values that might differ from the nemesis block file are ``identifier``,  ``publicKey``, ``generationHash``, ``currencyMosaicId``, ``harvestMosaicId``.
+
+Generate the nemesis block
+---
+
+```ssh
+cd bin
+./catapult.tools.nemgen --nemesisProperties ../resources/mijin-test.properties
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:138]: why are you running nemgen twice?

File: docs/RUNNETWORKLIN.md [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+host = <YOUR_NODE_IP>
+friendlyName = myNode
+version = 0
+roles = Peer
+```
+
+2\. Open the ``config-user.properties`` and replace ``bootPrivateKey`` with a new private key for this node. The node's account is used by other peers to identify the node.
+
+NOTE: You can generate new accounts with the command ``./bin/catapult.tools.address -g 10 --network mijin-test``.
+
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:162]: this is also same as running node, and should be consolidated imo

File: docs/RUNNETWORKLIN.md [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+isAutoHarvestingEnabled = true
+...
+```
+
+Launch the network
+---
+
+```ssh
+cd bin
+./catapult.server
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:182]: likewise, this is probably also relevant to running node guide

File: docs/RUNNETWORKLIN.md (hunk 4 above)
Comment (dgarcia360) [reply docs/RUNNETWORKLIN.md:30]: Meaning "NOT recommended for production networks", right? I'll add a note. Do we have an alternative templates with the recommended specs?

File: README.md (hunk 2 above)
Comment (dgarcia360) [reply README.md]: Not dead, it's referenced at the end of the file:

`[Service Bootstrap]: https://github.com/techbureau/catapult-service-bootstrap`

**Comment by dgarcia360** [comment]
Thanks for the comments @Jaguar0625. I will move everything related to peer nodes configuration to the next guide.

File: docs/RUNNETWORKLIN.md (hunk 7 above)
Comment (dgarcia360) [reply docs/RUNNETWORKLIN.md:125]: nemgen runs twice to add the generated mosaic ids from the first execution to the ``config-network.properties`` file.

See https://github.com/tech-bureau/catapult-service-bootstrap/blob/master/cmds/bootstrap/dockerfiles/nemgen/userconfig/nemgen.sh#L18-L27

File: README.md [hunk 10]
```diff
@@ -1,21 +1,21 @@
...
-symbol-server code is organized as follows:
+catapult-server code is organized as follows:
 
 | Folder name | Description |
 | -------------|--------------|
-|/extensions | Modules that add features to the bare symbol-server. These capabilities are all optional because none of them impact consensus. |
+|/extensions | Modules that add features to the bare catapult-server. These capabilities are all optional because none of them impact consensus. |
 | /external | External dependencies that are built with the server. |
 | /plugins | Modules that introduce new and different ways to alter the chain's state via transactions. |
 |/resources | Default properties. These can be configured per network and node. |
```
Comment (Jaguar0625) [inline README.md:21]: you seem to have reverted spacing after first `|`

File: docs/RUNNETWORKLIN.md (hunk 4 above)
Comment (Jaguar0625) [reply docs/RUNNETWORKLIN.md:14]: @CrackTheCode016 or @kaiyzen might

File: docs/RUNNETWORKLIN.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+
+```ini
+[cpp]
+
+cppFileHeader = 
+
+[output]
+
+cppFile = 
+binDirectory = ../seed/network-test
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:60]: please remove trailing spaces

File: docs/RUNNETWORKLIN.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+```
+4\. Continue editing the nemesis block to fit the network requirements before moving to the next step.
+
+NOTE: Do not add the ``nemesisSignerPrivateKey`` account to the distribution list.
+The nemesis signer account will not be able to announce transactions.
+The mosaics received by the nemesis account will be lost forever.
+
+## Create the seed directory
+
+Create a directory to save the nemesis block compiled.
```
Comment (Jaguar0625) [inline docs/RUNNETWORKLIN.md:84]: ... generated nemesis block ?

**Comment by Jaguar0625** [comment]
fixed one typo, rebased, squashed and merged

**Pull request merged**